packages/vm/src/compatability.rs

```diff
@@ -347,4 +373,103 @@ mod test {
             Ok(_) => panic!("Didn't reject wasm with invalid api"),
         }
     }
+
+    #[test]
+    fn check_wasm_features_ok() {
+        let wasm = wat2wasm(
+            r#"(module
+            (type (func))
+            (func (type 0) nop)
+            (export "requires_water" (func 0))
```

Comment:
okay, so this checks that if only matches `requires_(w+)` statements, right, and eg. `requires_` and `REQUIRES_air` are ignored. Seems a good rule.

Reply:
Right, the idea is a feature must not be empty and the all processing here is case sensitive (which is right now not tested for the values)

Reply:
590466a tests that feature matching is case-sensitive (supporting "Water" does not fulfil the "water" requirement)